### ISSUE TITLE
error when training sample does not contain all labels

### ISSUE THREAD

**Issue by 3kumar** [pystruct/pystruct#185]
I am doing n-fold CV over my training set, so suppose if my training dataset in a fold does not all the labels, then i get an error

 self._train_seq[self._train_phase][1](*args, **kwargs)
  File "pyStruct_crf_node.py", line 30, in _stop_training
    self.ssvm.fit(self.X_train,self.y_train)
  File "/usr/local/lib/python2.7/dist-packages/pystruct/learners/frankwolfe_ssvm.py", line 288, in fit
    self.model.initialize(X, Y)
  File "/usr/local/lib/python2.7/dist-packages/pystruct/models/chain_crf.py", line 75, in initialize
    % (self.n_states, n_states))
ValueError: Expected 36 states, got 29

Why it is necessary to have all the labels for the training? What if change 'initialize=False' in the fit method?
My code is as follows:- 

model = ChainCRF(n_states=classes)
ssvm = FrankWolfeSSVM(model=model, C=.1, max_iter=10)
ssvm.fit(x_train,y_train)

**Comment by amueller** [comment]
yeah that's kind of expected but should be fixed. thanks for pointing it out.

**Comment by CgGraphic** [comment]
Any solution？ I got this problem too
